executor/new_executor.go

```diff
@@ -47,6 +47,7 @@ type HashJoinExec struct {
 	leftSmall    bool
 	matchedRows  []*Row
 	cursor       int
+	targetTypes  []*types.FieldType
```

Comment:
if join a.id = b.id when a.id is an int and b.id is a decimal, they will cast to same type before getting hash key.